amp/WEB-INF/src/org/digijava/module/aim/util/SectorUtil.java

```diff
@@ -1214,12 +1238,109 @@ public static List<AmpActivityVersion> getActivitiesForSector(Long id) {
         }
         return activities;
     }
-    
+
     public static List<String> getAllSectorColumnNames() {
         // not much benefit from generating names
         return Arrays.asList(
                 ColumnConstants.PRIMARY_SECTOR, ColumnConstants.PRIMARY_SECTOR_SUB_SECTOR, ColumnConstants.PRIMARY_SECTOR_SUB_SUB_SECTOR,
                 ColumnConstants.SECONDARY_SECTOR, ColumnConstants.SECONDARY_SECTOR_SUB_SECTOR, ColumnConstants.SECONDARY_SECTOR_SUB_SUB_SECTOR,
                 ColumnConstants.TERTIARY_SECTOR, ColumnConstants.TERTIARY_SECTOR_SUB_SECTOR, ColumnConstants.TERTIARY_SECTOR_SUB_SUB_SECTOR);
     }
+
+    //region Sector Mapping
+    /**
+     * Returns all sector mappings
+     */
+    public static Collection getAllSectorMappings() {
+        String queryString = null;
+        Session session = null;
+        Collection col = null;
+        Query qry = null;
+
+        try {
+            session = PersistenceManager.getSession();
+            queryString = "select asm from " + AmpSectorMapping.class.getName() + " asm";
+            qry = session.createQuery(queryString);
+            col = qry.list();
+            session.flush();
```

Comment:
Since we are getting this again,I was wondering what get if we dont flush after a select operation. Because typically we should not flush after a select since there is no use for synchronizing the persistence context.. I think we should check this and rectify it.

Reply:
done